packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2146,6 +2159,38 @@ export class KeyringController extends BaseController<
     return serializedKeyrings;
   }
 
+  /**
+   * Serialize the current array of keyring instances and their metadata,
+   * including unsupported keyrings by default.
+   *
+   * @param options - Method options.
+   * @param options.includeUnsupported - Whether to include unsupported keyrings.
+   * @returns The serialized keyrings.
+   */
+  async #getSerializedKeyringsAndMetadata(
+    { includeUnsupported }: { includeUnsupported: boolean } = {
+      includeUnsupported: true,
+    },
+  ): Promise<SerializedKeyringAndMetadata[]> {
+    const serializedKeyrings = await this.#getSerializedKeyrings({
+      includeUnsupported,
+    });
+
+    const serializedKeyringsAndMetadata: SerializedKeyringAndMetadata[] =
+      serializedKeyrings.map((serialized, index) => {
+        return {
+          ...serialized,
+          metadata: this.#keyringsMetadata[index],
```

Comment:
hmm, will this metadata be deserialized into keyring instances as well? not sure if we want that

Reply:
yes, indeed we don't want this 🤔 I'll add another indirection here, like:
```ts
  return {
    keyring: serialized,
    metadata: this.#keyringsMetadata[index],
  };
```

I thought I did that initially 🙈 thanks for catching this!

Reply:
Though, as this function is mainly used for things that don't need metadata (like keyring restore), do you think we could move this into `#persistAllKeyrings` directly, or into another method specific to check if we need to update the controller state/vault?